src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs

```diff
@@ -149,21 +150,46 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
             return true;
         }
 
-        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, [NotNullWhen(true)] out string? fileContents)
+        private static bool TryReadFile(string path, [NotNullWhen(true)] out string? contents)
         {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+
+            byte[] bytes = ArrayPool<byte>.Shared.Rent(4096);
+            int count = 0;
+
             try
             {
-                using (var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                using var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false);
```

Comment:
@jkotas, regarding FileStream, one thing I noticed while working on this was we rely on Length property in other places such as this: https://github.com/dotnet/runtime/blob/a2ba504a8d696a5984464cfdd4fce3ea6cd4ed8c/src/libraries/System.Private.CoreLib/src/Internal/IO/File.cs#L52-L57 which is called by https://github.com/dotnet/runtime/blob/a2ba504a8d696a5984464cfdd4fce3ea6cd4ed8c/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs#L184 but here I was getting `Length: 0` (no exception which property can throw; e.g. /proc/* has CanSeek=True). Could you please give a pointer why is that? Moreover, (assuming it is a bug) would it make sense to make that internal/File.cs independent of fs.Length property?

Reply:
internal/File.cs is used to read timezone files that are regular files. I do not see a problem here.

Reply:
Ah, ok. Length property checks the stat and on Linux, stat returns 0 for files under /proc (maybe because it requires read(2) call to populate data):
```
$ stat /proc/self/stat
  File: /proc/self/stat
  Size: 0         	Blocks: 0          IO Block: 1024   regular empty file
...
```